src/firebase/models/players_in_room.js

```diff
@@ -0,0 +1,34 @@
+import { ref, onValue, onDisconnect, set, child } from "firebase/database";
+import rtdb from "../rtdb";
+
+const rooms = ref(rtdb, "rooms");
+const amOnline = ref(rtdb, ".info/connected");
+
+function createNewRandomUserRef(routeRoomID) {
+  const userId =
+    "rooms/" +
+    routeRoomID +
+    "/" +
+    Math.trunc(Math.random() * 100000).toString();
+
+  return ref(rtdb, userId);
+}
```

Comment:
Since we aren't creating actual users, we just create some random temporary users in the database and assign them a user id. The user id is put into the db using the routeRoomID to construct it (which keeps the user connected to a room and also ensures a clean namespace) and also adds a random number from 0 to 100000 and then makes it a string. The function then returns the userId as a reference so it can be used elsewhere